lib/dry/configurable.rb

```diff
@@ -48,8 +48,8 @@ def inherited(subclass)
     #
     # @api public
     def config
+      return @_config if defined?(@_config)
       @_config_mutex.synchronize do
-        return @_config if defined?(@_config)
```

Comment:
Good catch, but you can't just remove this line, the result won't be equivalent, consider this:
thread 1
@_config not defined, move on
acquire lock

thread 2
@_config not defined, move on
wait on mutex

thread 1
set @_config
release mutex

thread 2
acquire mutex
set @_config <- NOT SAFE
release mutex

So, along your return statement you should change `@_config = ` with `@_config ||=`